vertical-pod-autoscaler/updater/apimock/vpa_api_mock.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package apimock contains temporary definitions of Vertical Pod Autoscaler related objects - to be replaced with real implementation
+// Definitions based on VPA design doc : https://github.com/kubernetes/community/pull/338
+package apimock
+
+import (
+	"math/rand"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	apiv1 "k8s.io/kubernetes/pkg/api/v1"
+)
+
+// VerticalPodAutoscaler Represents Vertical Pod Autoscaler configuration - to be replaced by real implementation
+type VerticalPodAutoscaler struct {
+	// Specification
+	Spec Spec
+	// Current state of VPA
+	Status Status
+}
+
+// Spec holds Vertical Pod Autoscaler configuration
+type Spec struct {
+	// Defines which pods this Autoscaler should watch and update
+	Target Target
+	// Policy for pod updates
+	UpdatePolicy UpdatePolicy
+	// Policy for container resources updates
+	ResourcesPolicy ResourcesPolicy
+}
+
+// Status holds current Vertical Pod Autoscaler state
+type Status struct {
+	// Recommended resources allocation
+	Recommendation *Recommendation
+}
+
+// Target Specifies pods to be managed by Vertical Pod Autoscaler
+type Target struct {
+	// Pod labels to match
+	MatchLabels map[string]string
```

Comment:
FYI. we will be using label selectors to match pods. You may consider already implementing it, if you have time.
See how it's defined and used in the Horizontal Pod Autoscaler:
- https://github.com/kubernetes/kubernetes/blob/9ae2dfacf196ca7dbee798ee9c3e1663a5f39473/pkg/apis/autoscaling/v1/types.go#L133
- https://github.com/kubernetes/kubernetes/blob/73fb978181dcb2e1aa54d09cbd0c48826f63d39b/pkg/controller/podautoscaler/replica_calculator.go#L54